### PR TITLE
[FAB-18235] Properly handle Newest when a peer is bootstrapped from a snapshot

### PR DESCRIPTION
**Pull request by wenjianqiao**
Signed-off-by: Wenjian Qiao <wenjianq@gmail.com>

#### Type of change
- New feature

#### Description
- Update block iterator generation code to properly handle Newest for snapshot

#### Related issues
https://jira.hyperledger.org/browse/FAB-18235

### CODE REVIEW

File: common/ledger/blockledger/fileledger/impl_test.go [hunk 1]
```diff
@@ -243,6 +246,73 @@ func TestBlockedRetrieval(t *testing.T) {
...
+
+	b3 := blockledger.CreateNextBlock(fl, []*cb.Envelope{envelope})
+	fl.Append(b3)
+
+	block, status = it.Next()
+	require.Equal(t, cb.Status_SUCCESS, status)
+	require.Equal(t, uint64(3), block.Header.Number)
+}
+
+// TODO
```
Comment (manish-sethi) [inline common/ledger/blockledger/fileledger/impl_test.go]: leftover

**Review by manish-sethi — CHANGES_REQUESTED** [review]
LGTM, just one minor suggestion about package name

File: common/ledger/blkstorage/test_exports.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package blkstorage
+
```
[the remark this answers is not in the document]
Comment (wenjianqiao) [reply common/ledger/blkstorage/test_exports.go]: Done

**Review by manish-sethi — DISMISSED** [review]

**Comment by manish-sethi** [comment]
@wenjianqiao - I guess that this PR can be merged as is if you want. Just update the commit message to reflect the change.

**Comment by wenjianqiao** [comment]
> @wenjianqiao - I guess that this PR can be merged as is if you want. Just update the commit message to reflect the change.

@manish-sethi Yes, it is safe to merge the PR as is. Meanwhile, I had to undo the change in internal/peer/chaincode/common.go (i.e., didn't change from Newest to NextCommit for chaincode invocation), because the change caused test flake. Still investigating why.

**Comment by wenjianqiao** [comment]
@manish-sethi Found why IT tests failed with NextCommit option. In a distributed env, some peers may get block height after the next new block is committed. In this case, using NextCommit option will never get the block it expects. Therefore, we should keep using Newest option when waiting for next committed block. The PR is done and can be merged.

**Comment by manish-sethi** [comment]
> @manish-sethi Found why IT tests failed with NextCommit option. In a distributed env, some peers may get block height after the next new block is committed. In this case, using NextCommit option will never get the block it expects. Therefore, we should keep using Newest option when waiting for next committed block. The PR is done and can be merged.

Thanks @wenjianqiao for investigating it. As this could in general be a problem with `Newest` option as well. Can you open a Jira for this. As far as the scope of this PR is concerned, we are good with the changes.

**Review by manish-sethi — APPROVED** [review]

**Pull request merged**